documentation/trunk/solutions/basic-computing/eiffelbase/eiffelbase-tutorial/eiffelbase-kernel.wiki

```diff
@@ -102,7 +102,6 @@ These two features enable clients to ascertain the dynamic type of an entity at
 
 ==Miscellaneous==
 
-The query Void, of type <eiffel>NONE</eiffel>, denotes a reference that is always void - not attached to any object. <br/>
```

Comment:
Hi! Sorry for the delay, I'm currently very busy...

Ok I recovers the initial state.
However, I'm not sure that it is exact since the emergence of void-safety.
Void-safety introduced an inconsistency in the Eiffel type hierarchy, but I think so it is fixable ;)

Indeed, if ```Void``` is a value of ```NONE``` and ```NONE``` is the descendant of every class, then ```Void``` should be assignable to everything. But it is not acceptable.

Thus, we need to remove one of the next constraints:
- ```Void``` is  a value of ```NONE``` (```detachable NONE``` is equivalent to ```NONE```)
- ```NONE``` is the descendant of every class

If the second constraint is removed, this implies several inconsistencies:
- in exportation clauses
- in inspect clauses (type intervals)

Therefore, the removal of the first seems more convenient.

I imagine ```detachable TYPE``` as a union type ```Void | TYPE```.
Thus ```detachable NONE``` would be ```Void | NONE```.
In this case ```Void``` is no longer a valid value of ```NONE```. ```NONE``` has no valid values and instances, it is just the (frozen) descendant of every class.